insights/core/spec_factory.py

```diff
@@ -80,7 +80,8 @@ def content(self):
 
         if self._content is None:
             try:
-                self._content = self.load()
+                if not self.loaded:
```

Comment:
This is a bit of a nitpick, but could we add this check to the if statement on line 81 above?

Reply:
Makes sense, modified. Please check again